core/src/message/context.rs

```diff
@@ -278,20 +278,26 @@ impl MessageContext {
         packet: InitPacket,
         delay: u32,
     ) -> Result<(MessageId, ProgramId), Error> {
-        let program_id = packet.destination();
-
-        if self.store.initialized.contains(&program_id) {
-            return Err(Error::DuplicateInit);
-        }
-
         let last = self.store.outgoing.len() as u32;
 
         if last >= self.settings.outgoing_limit {
             return Err(Error::OutgoingMessagesAmountLimitExceeded);
         }
 
         let message_id = MessageId::generate_outgoing(self.current.id(), last);
-        let message = InitMessage::from_packet(message_id, packet);
+
+        let message =
+            InitMessage::from_packet(message_id, packet, Some(self.store.initialized.len()));
+
+        let program_id = message.destination();
+
+        if program_id == ProgramId::SYSTEM {
```

Comment:
you don't need that check
and especially - here

Reply:
It was a check that existed previously, I can remove it if you believe it is not needed

Reply:
it's no longer needed so unreachable may be put on journal notes processing level but I don't see the real profit in it